File: JS/LeetC/g75/longestPalindrome.js
function longestPalindrome(s) {
  const set = new Set();

  let count = 0;
  for (let c of s) {
    if (set.has(c)) {
      count += 2;
      set.delete(c)
    } else {
      set.add(c);
    }
  }

  return set.size > 0 ? count + 1 : count;
}

File: JS/LeetC/g75/longestPalindrome.py
def longestPalindrome(s):
    sset = set()

    count = 0
    for letter in s:
        if letter in sset:
            count += 2
            sset.remove(letter)
        else:
            sset.add(letter)

    return count + 1 if len(sset) > 0 else count
